FEM/rf_num_new.cpp

```diff
@@ -698,6 +702,17 @@ ios::pos_type CNumerics::Read(ifstream* num_file)
 			     << "\n";
 			continue;
 		}
+		// automatic damping of Newton scheme
+		if (line_string.find("$AUTOMATIC_NEWTON_DAMPING") != string::npos) // NW
+		{
+			line.str(GetLineFromFile1(num_file));
+			line >> automatic_damping_tolerance; //if NR error decreases by less than this factor, the next step will be dampened
```

Comment:
Yes, the damping should be determined by error. However, the damping factor has to be specified in advance. Actually, the tolerance of automatic_damping_tolerance should be the same as to that of the Newton such that the damping is applied when a Newton step error fails in quadratic convergence (in the strict manner) or the error increases.

If  automatic_damping is used, one may think that he only need to give the keyword to enable the method. Anything else including when damping, and the value of damping are conducted by the program  automatically (e.g by linear search).